### PR TITLE
NFDIV-2777 Prioritise submitted cases

### PR DESCRIPTION
**Pull request by lara-mpt**
### Change description ###

Prioritise submitted cases. Return latest case which is not in a draft state.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-2777

### CODE REVIEW

File: src/main/app/case/case-api.ts [hunk 1]
```diff
@@ -76,6 +84,26 @@ export class CaseApi {
     return false;
   }
 
+  private getLatestPrioritisedUserCase(userCases: CcdV1Response[] | false): CaseWithId | false {
+    const isDraftState = [
```
Comment (benouaer) [inline src/main/app/case/case-api.ts:88]: the variable name suggests it's a boolean, but it's an array

Comment (danielbe11) [inline src/main/app/case/case-api.ts:88]: preSubmissionStates?

File: src/main/app/case/case-api.ts [hunk 2]
```diff
@@ -76,6 +84,26 @@ export class CaseApi {
...
+      State.Draft,
+      State.AwaitingApplicant1Response,
+      State.AwaitingApplicant2Response,
+      State.Applicant2Approved,
+    ];
+    if (userCases && userCases.length > 1) {
+      const submittedUserCase = userCases.slice(1).find(userCase => !isDraftState.includes(userCase.state));
+      if (isDraftState.includes(userCases[0].state) && submittedUserCase) {
```
Comment (benouaer) [inline src/main/app/case/case-api.ts]: why is the 0 case treated differently?

Comment (lara-mpt) [reply src/main/app/case/case-api.ts]: @benouaer I've just updated that :)

**Review by benouaer — APPROVED** [review]

**Review by danielbe11 — APPROVED** [review]

File: src/main/app/case/case-api.test.ts [hunk 3]
```diff
@@ -172,6 +172,68 @@ describe('CaseApi', () => {
...
+    const mockSubmittedCase = {
+      id: '1',
+      state: State.Submitted,
+      case_data: { divorceOrDissolution: DivorceOrDissolution.DIVORCE },
+    };
+    mockApiClient.findExistingUserCases.mockResolvedValue([mockDraftCase, mockSubmittedCase]);
+
+    const userCase = await api.getExistingUserCase(DivorceOrDissolution.DIVORCE);
+
+    expect(userCase).toStrictEqual({
```
Comment (benouaer) [inline src/main/app/case/case-api.test.ts:190]: why not just put mockSubmittedCase

File: src/main/app/case/case-api.test.ts [hunk 4]
```diff
@@ -172,6 +172,68 @@ describe('CaseApi', () => {
...
+      id: '1',
+      state: State.Submitted,
+      divorceOrDissolution: DivorceOrDissolution.DIVORCE,
+    });
+  });
```
Comment (benouaer) [inline src/main/app/case/case-api.test.ts:195]: why not just put mockSubmittedCase

File: src/main/app/case/case-api.test.ts [hunk 5]
```diff
@@ -172,6 +172,68 @@ describe('CaseApi', () => {
...
+    mockApiClient.findExistingUserCases.mockResolvedValue([mockDraftCase, mockSubmittedCase]);
+
+    const userCase = await api.getExistingUserCase(DivorceOrDissolution.DIVORCE);
+
+    expect(userCase).toStrictEqual({
+      id: '1',
+      state: State.AwaitingApplicant2Response,
+      divorceOrDissolution: DivorceOrDissolution.DIVORCE,
+    });
+  });
```
Comment (benouaer) [inline src/main/app/case/case-api.test.ts:217]: why not just put mockSubmittedCase

File: src/main/app/case/case-api.test.ts [hunk 6]
```diff
@@ -172,6 +172,68 @@ describe('CaseApi', () => {
...
+    const mockDraftCase2 = {
+      id: '2',
+      state: State.Draft,
+      case_data: { divorceOrDissolution: DivorceOrDissolution.DIVORCE },
+    };
+    mockApiClient.findExistingUserCases.mockResolvedValue([mockDraftCase, mockDraftCase2]);
+
+    const userCase = await api.getExistingUserCase(DivorceOrDissolution.DIVORCE);
+
+    expect(userCase).toStrictEqual({ id: '1', state: State.Draft, divorceOrDissolution: DivorceOrDissolution.DIVORCE });
```
Comment (benouaer) [inline src/main/app/case/case-api.test.ts:234]: why not just put mockDraftCase

**Review by benouaer — APPROVED** [review]

File: src/main/app/case/case-api.test.ts (hunk 4 above)
Comment (lara-mpt) [reply src/main/app/case/case-api.test.ts:195]: you can't because the format is returned differently - I just copied the format of the other tests

**Pull request merged**